qase-api/src/main/java/io/qase/api/CasesStorage.java

```diff
@@ -0,0 +1,42 @@
+package io.qase.api;
+
+import io.qase.client.model.ResultCreate;
+
+public final class CasesStorage {
+
+    private static final ThreadLocal<ResultCreate> CURRENT_CASE = new ThreadLocal<>();
+
+    public static void startCase(ResultCreate resultCreate) {
+        ensureCaseIsNotInProgress();
+
+        CURRENT_CASE.set(resultCreate);
+    }
+
+    public static void stopCase() {
+        ensureCaseIsInProgress();
+
+        CURRENT_CASE.remove();
+    }
+
+    public static ResultCreate getCurrentCase() {
+        ensureCaseIsInProgress();
+
+        return CURRENT_CASE.get();
+    }
+
+    public static boolean isCaseInProgress() {
+        return CURRENT_CASE.get() != null;
+    }
+
+    private static void ensureCaseIsInProgress() {
+        if (!isCaseInProgress()) {
+            throw new IllegalStateException("A case has not been started yet.");
```

Comment:
I think it would be better if we use error logging instead of throwing exception. We must not break the test using our library

Reply:
I agree with you.
However, wouldn't it be handy if the library hinting a users about they had not notified  QaseTestCaseListener about a test case life cycle changed?

Reply:
Can we do it through logs?

Reply:
Yes, sure